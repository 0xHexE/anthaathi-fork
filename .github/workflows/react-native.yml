name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
