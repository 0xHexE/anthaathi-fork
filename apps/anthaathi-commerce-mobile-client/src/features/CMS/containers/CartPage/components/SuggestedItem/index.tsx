import * as React from 'react';
import {Button, Card, Text, Title} from 'react-native-paper';
import {Image, Pressable, View, VirtualizedList} from 'react-native';
import Entypo from 'react-native-vector-icons/Entypo';
import {useResponsiveValue} from '../../../../utils/useResponsiveValue';
import {useIntl} from 'react-intl';
import {CartPageComponentType} from '../../../../types/common';

export interface ProductProps {
  name: string;
  description?: string;
  price: number;
  currency: string;
  image: string;
  weight_unit: string;
  packaging: string;
  key: string;
  notes?: string;
}

export interface FeaturedCollectionProps {
  title: string;
  products: ProductProps[];
  handlePress?: () => void; // view all product link
}

export default function SuggestedItem({
  title,
  products,
  handlePress,
}: FeaturedCollectionProps) {
  const intl = useIntl();
  const itemHeight = useResponsiveValue([140, 180, 250, 290]);
  const itemWidth = useResponsiveValue([130, 170, 240, 280]);

  return (
    <View testID="suggestedItem">
      <View
        style={{
          flexDirection: 'row',
          justifyContent: 'space-between',
          alignItems: 'center',
          marginHorizontal: 10,
        }}>
        <Text
          variant="titleLarge"
          style={{marginBottom: 9, fontSize: 18, color: '#808080'}}>
          {title}
        </Text>
        <Pressable onPress={handlePress} testID="handlePressSuggestedItem">
          <Text
            variant="titleMedium"
            style={{
              marginBottom: 9,
              textDecorationLine: 'underline',
              fontSize: 14,
            }}>
            {intl.formatMessage({defaultMessage: 'View All'})}
          </Text>
        </Pressable>
      </View>

      <View>
        <VirtualizedList<ProductProps>
          testID="suggestedItemList"
          data={products}
          initialNumToRender={4}
          horizontal
          renderItem={({item}) => (
            <ItemRenderer
              item={item}
              itemHeight={itemHeight}
              itemWidth={itemWidth}
            />
          )}
          getItemCount={() => products.length}
          keyExtractor={item => item.key}
          getItem={(res, index) => res[index]}
        />
      </View>
    </View>
  );
}

function ItemRenderer({
  item,
  itemHeight,
  itemWidth,
}: {
  item: ProductProps;
  itemHeight: number;
  itemWidth: number;
}) {
  const intl = useIntl();
  return (
    <View
      style={{
        marginVertical: 5,
        marginHorizontal: 10,
      }}
      key={item.key}>
      <Card.Content style={{alignItems: 'center'}}>
        <Image
          style={{height: itemHeight, width: itemWidth, borderRadius: 4}}
          source={{
            uri: item.image,
          }}
        />
        <View
          style={{
            flexDirection: 'row',
            alignItems: 'center',
          }}>
          <Title style={{fontSize: 14, marginRight: 10, fontWeight: 'bold'}}>
            {item.name}
          </Title>
          <Entypo name="info-with-circle" color="#364A15" size={18} />
        </View>
        <View
          style={{
            flexDirection: 'row',
            marginTop: 5,
            alignItems: 'center',
            justifyContent: 'center',
          }}>
          <Text style={{color: '#008D3E', fontSize: 14, fontWeight: '400'}}>
            {intl.formatNumber(item.price, {
              style: 'currency',
              currency: item.currency,
            })}
          </Text>
          <Text style={{color: '#808080', fontSize: 12, fontWeight: '400'}}>
            {' / ' + item.packaging}
          </Text>
        </View>

        <Text style={{color: '#808080', fontSize: 12, fontWeight: '400'}}>
          {item.notes}
        </Text>

        <Button
          mode="contained"
          style={{backgroundColor: '#F1F9F4', width: '100%', borderRadius: 0}}
          labelStyle={{color: '#008D3E'}}
          onPress={() => console.log('Pressed')}>
          Add
        </Button>
      </Card.Content>
    </View>
  );
}

export const SuggestedItemCMSInput = {
  _component: CartPageComponentType.SuggestedItem,
  component: SuggestedItem,
};
