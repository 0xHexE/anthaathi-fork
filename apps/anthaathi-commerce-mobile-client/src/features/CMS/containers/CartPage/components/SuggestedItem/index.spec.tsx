import React from 'react';
import 'react-native';
import {fireEvent, render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import SuggestedItem from './index';

describe('SuggestedItem', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <SuggestedItem
            title="Suggested"
            products={[
              {
                name: 'Baby Yellow Pepper',
                image:
                  'https://burst.shopifycdn.com/photos/fruit-plate.jpg?width=373&height=373&format=pjpg&exif=1&iptc=1',
                key: '12',
                price: 12,
                currency: 'USD',
                weight_unit: 'KG',
                packaging: 'pack',
              },
              {
                name: 'Capsicum mixed',
                image:
                  'https://burst.shopifycdn.com/photos/red-and-green-gooseberries-against-white.jpg?width=373&format=pjpg&exif=1&iptc=1',
                key: '23',
                price: 23,
                currency: 'USD',
                weight_unit: 'KG',
                packaging: 'pack',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('suggestedItem')).toBeTruthy();
  });

  it('renders expected number of products', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <SuggestedItem
            title="Suggested"
            products={[
              {
                name: 'Baby Yellow Pepper',
                image:
                  'https://burst.shopifycdn.com/photos/fruit-plate.jpg?width=373&height=373&format=pjpg&exif=1&iptc=1',
                key: '12',
                price: 12,
                currency: 'USD',
                weight_unit: 'KG',
                packaging: 'pack',
              },
              {
                name: 'Capsicum mixed',
                image:
                  'https://burst.shopifycdn.com/photos/red-and-green-gooseberries-against-white.jpg?width=373&format=pjpg&exif=1&iptc=1',
                key: '23',
                price: 23,
                currency: 'USD',
                weight_unit: 'KG',
                packaging: 'pack',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    const getTestId = temp.queryByTestId('suggestedItemList')!;
    expect(getTestId.props.data.length).toBe(2);
    expect(getTestId.props.data[0]).toEqual({
      name: 'Baby Yellow Pepper',
      image:
        'https://burst.shopifycdn.com/photos/fruit-plate.jpg?width=373&height=373&format=pjpg&exif=1&iptc=1',
      key: '12',
      price: 12,
      currency: 'USD',
      weight_unit: 'KG',
      packaging: 'pack',
    });
  });

  it('should have SuggestedItem title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <SuggestedItem
            title="Suggested"
            products={[
              {
                name: 'Capsicum mixed',
                image:
                  'https://burst.shopifycdn.com/photos/red-and-green-gooseberries-against-white.jpg?width=373&format=pjpg&exif=1&iptc=1',
                key: '23',
                price: 23,
                currency: 'USD',
                weight_unit: 'KG',
                packaging: 'pack',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );
    expect(temp.queryByText('Suggested')).toBeTruthy();
  });

  it('should call when we call tap handlePress', function () {
    const onpress = jest.fn();

    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <SuggestedItem
            title="Suggested"
            handlePress={onpress}
            products={[
              {
                name: 'Capsicum mixed',
                image:
                  'https://burst.shopifycdn.com/photos/red-and-green-gooseberries-against-white.jpg?width=373&format=pjpg&exif=1&iptc=1',
                key: '23',
                price: 23,
                currency: 'USD',
                weight_unit: 'KG',
                packaging: 'pack',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    fireEvent.press(temp.queryByTestId('handlePressSuggestedItem')!);
    expect(onpress).toBeCalledTimes(1);
  });
});
