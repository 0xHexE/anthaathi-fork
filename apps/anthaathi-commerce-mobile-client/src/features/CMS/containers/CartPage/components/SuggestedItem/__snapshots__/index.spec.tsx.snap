// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestedItem should render the item 1`] = `
<View
  testID="suggestedItem"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(28, 27, 31, 1)",
            "fontSize": 22,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 28,
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "writingDirection": "ltr",
          },
          Object {
            "color": "#808080",
            "fontSize": 18,
            "marginBottom": 9,
          },
        ]
      }
    >
      Suggested
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="handlePressSuggestedItem"
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 16,
              "fontWeight": "500",
              "letterSpacing": 0.15,
              "lineHeight": 24,
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "writingDirection": "ltr",
            },
            Object {
              "fontSize": 14,
              "marginBottom": 9,
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        View All
      </Text>
    </View>
  </View>
  <View>
    <RCTScrollView
      data={
        Array [
          Object {
            "currency": "USD",
            "image": "https://burst.shopifycdn.com/photos/fruit-plate.jpg?width=373&height=373&format=pjpg&exif=1&iptc=1",
            "key": "12",
            "name": "Baby Yellow Pepper",
            "packaging": "pack",
            "price": 12,
            "weight_unit": "KG",
          },
          Object {
            "currency": "USD",
            "image": "https://burst.shopifycdn.com/photos/red-and-green-gooseberries-against-white.jpg?width=373&format=pjpg&exif=1&iptc=1",
            "key": "23",
            "name": "Capsicum mixed",
            "packaging": "pack",
            "price": 23,
            "weight_unit": "KG",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={4}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      testID="suggestedItemList"
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Object {
                "marginHorizontal": 10,
                "marginVertical": 5,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "paddingHorizontal": 16,
                  },
                  undefined,
                  Object {
                    "alignItems": "center",
                  },
                ]
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://burst.shopifycdn.com/photos/fruit-plate.jpg?width=373&height=373&format=pjpg&exif=1&iptc=1",
                  }
                }
                style={
                  Object {
                    "borderRadius": 4,
                    "height": 180,
                    "width": 170,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "rgba(28, 27, 31, 0.87)",
                          "writingDirection": "ltr",
                        },
                        Array [
                          Object {
                            "fontSize": 20,
                            "letterSpacing": 0.15,
                            "lineHeight": 30,
                            "marginVertical": 2,
                          },
                          Object {
                            "fontSize": 14,
                            "fontWeight": "bold",
                            "marginRight": 10,
                          },
                        ],
                      ],
                    ]
                  }
                >
                  Baby Yellow Pepper
                </Text>
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#364A15",
                        "fontSize": 18,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Entypo",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  Ôáç
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginTop": 5,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "color": "#008D3E",
                        "fontSize": 14,
                        "fontWeight": "400",
                      },
                    ]
                  }
                >
                  $12.00
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "color": "#808080",
                        "fontSize": 12,
                        "fontWeight": "400",
                      },
                    ]
                  }
                >
                   / pack
                </Text>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#808080",
                      "fontSize": 12,
                      "fontWeight": "400",
                    },
                  ]
                }
              />
              <View
                collapsable={false}
                style={
                  Object {
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "backgroundColor": "#F1F9F4",
                        "borderColor": "transparent",
                        "borderRadius": 0,
                        "borderStyle": "solid",
                        "borderWidth": 0,
                        "minWidth": 64,
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      accessibilityRole="button"
                      accessibilityState={
                        Object {
                          "disabled": false,
                        }
                      }
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Object {
                            "borderRadius": 20,
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "center",
                            },
                            undefined,
                          ]
                        }
                      >
                        <Text
                          numberOfLines={1}
                          selectable={false}
                          style={
                            Array [
                              Object {
                                "color": "rgba(28, 27, 31, 1)",
                                "fontSize": 14,
                                "fontWeight": "500",
                                "letterSpacing": 0.1,
                                "lineHeight": 20,
                              },
                              Object {
                                "textAlign": "left",
                              },
                              Object {
                                "writingDirection": "ltr",
                              },
                              Array [
                                Object {
                                  "marginHorizontal": 16,
                                  "marginVertical": 9,
                                  "textAlign": "center",
                                },
                                false,
                                Object {
                                  "marginHorizontal": 24,
                                  "marginVertical": 10,
                                },
                                undefined,
                                false,
                                Object {
                                  "color": "rgba(255, 255, 255, 1)",
                                  "fontFamily": "System",
                                  "fontSize": 14,
                                  "fontWeight": "500",
                                  "letterSpacing": 0.1,
                                  "lineHeight": 20,
                                },
                                Object {
                                  "color": "#008D3E",
                                },
                              ],
                            ]
                          }
                        >
                          Add
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Object {
                "marginHorizontal": 10,
                "marginVertical": 5,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "paddingHorizontal": 16,
                  },
                  undefined,
                  Object {
                    "alignItems": "center",
                  },
                ]
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://burst.shopifycdn.com/photos/red-and-green-gooseberries-against-white.jpg?width=373&format=pjpg&exif=1&iptc=1",
                  }
                }
                style={
                  Object {
                    "borderRadius": 4,
                    "height": 180,
                    "width": 170,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "rgba(28, 27, 31, 0.87)",
                          "writingDirection": "ltr",
                        },
                        Array [
                          Object {
                            "fontSize": 20,
                            "letterSpacing": 0.15,
                            "lineHeight": 30,
                            "marginVertical": 2,
                          },
                          Object {
                            "fontSize": 14,
                            "fontWeight": "bold",
                            "marginRight": 10,
                          },
                        ],
                      ],
                    ]
                  }
                >
                  Capsicum mixed
                </Text>
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#364A15",
                        "fontSize": 18,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Entypo",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  Ôáç
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginTop": 5,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "color": "#008D3E",
                        "fontSize": 14,
                        "fontWeight": "400",
                      },
                    ]
                  }
                >
                  $23.00
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "color": "#808080",
                        "fontSize": 12,
                        "fontWeight": "400",
                      },
                    ]
                  }
                >
                   / pack
                </Text>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#808080",
                      "fontSize": 12,
                      "fontWeight": "400",
                    },
                  ]
                }
              />
              <View
                collapsable={false}
                style={
                  Object {
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "backgroundColor": "#F1F9F4",
                        "borderColor": "transparent",
                        "borderRadius": 0,
                        "borderStyle": "solid",
                        "borderWidth": 0,
                        "minWidth": 64,
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      accessibilityRole="button"
                      accessibilityState={
                        Object {
                          "disabled": false,
                        }
                      }
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Object {
                            "borderRadius": 20,
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "center",
                            },
                            undefined,
                          ]
                        }
                      >
                        <Text
                          numberOfLines={1}
                          selectable={false}
                          style={
                            Array [
                              Object {
                                "color": "rgba(28, 27, 31, 1)",
                                "fontSize": 14,
                                "fontWeight": "500",
                                "letterSpacing": 0.1,
                                "lineHeight": 20,
                              },
                              Object {
                                "textAlign": "left",
                              },
                              Object {
                                "writingDirection": "ltr",
                              },
                              Array [
                                Object {
                                  "marginHorizontal": 16,
                                  "marginVertical": 9,
                                  "textAlign": "center",
                                },
                                false,
                                Object {
                                  "marginHorizontal": 24,
                                  "marginVertical": 10,
                                },
                                undefined,
                                false,
                                Object {
                                  "color": "rgba(255, 255, 255, 1)",
                                  "fontFamily": "System",
                                  "fontSize": 14,
                                  "fontWeight": "500",
                                  "letterSpacing": 0.1,
                                  "lineHeight": 20,
                                },
                                Object {
                                  "color": "#008D3E",
                                },
                              ],
                            ]
                          }
                        >
                          Add
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
